frontend/src/composables/use-sensitive-media.ts

```diff
@@ -0,0 +1,83 @@
+import { computed, watch } from "vue"
+
+import type { Media } from "~/types/media"
+
+import { useUiStore } from "~/stores/ui"
+
+type SensitiveVisibilityState =
+  | "insensitive"
+  | "sensitive-shown"
+  | "sensitive-hidden"
+
+/**
+ * A helper composable for working with sensitive media.
+ * Provides the current visibility of a sensitive media,
+ * along with toggles for hiding/revealing the media.
+ */
+export function useSensitiveMedia(
+  media: Pick<Media, "id" | "isSensitive"> | null
```

Comment:
Why do we want to allow `null` as a value for `media` here?
The code would be much simpler if we didn't :)

Reply:
Later, in the safety wall PR, we call useSensitiveMedia on the single result pages where the media might be `null`. I wasn't sure of the best way to handle that and this felt the easiest.